plugins/identity/app/services/service_layer/identity_services/credential.rb

```diff
@@ -5,21 +5,70 @@ module IdentityServices
     # This module implements Openstack Credential API
     module Credential
 
-      AWSCreds = Struct.new(:access, :secret, :error)
-
-      def aws_credentials(user_id, project_id)
-        aws_creds = [] 
-        response = elektron_identity.get("users/#{user_id}/credentials/OS-EC2")
-        if !response.body["credentials"].empty?
-          creds_hash = response.body["credentials"]
-          creds_hash.each do | creds |
-            next if creds['tenant_id'] != project_id
-            aws_creds << AWSCreds.new(creds["access"] , creds["secret"], nil) 
-          end
+      def credential_map
+        @credential_map ||= class_map_proc(Identity::Credential)
+      end
+
+      def new_credential(attributes = {})
+        credential_map.call(attributes)
+      end
+
+      def find_credential!(user_id = nil, access_id = nil, options = {})
```

Comment:
Please don't hardcode the credential type (OS-EC2) into the general credential API. Either pass the type in the params or rename the method to make it clear that this works only for EC2 credentials, so e.g. `find_ec2_credential`

Reply:
done.